services/core/shared/core/delegates.js

```diff
@@ -119,8 +132,113 @@ const getTotalNumberOfDelegates = async (params = {}) => {
 	return relevantDelegates.length;
 };
 
+const computeDelegateRankAndStatus = async () => {
+	const db = await pouchdb(config.db.collections.delegates.name);
+	const sdkVersion = Number(coreApi.getSDKVersion().split('sdk_v')[1]);
+	const numActiveForgers = 101;
+
+	const delegateStatus = {
+		ACTIVE: 'active',
+		STANDBY: 'standby',
+		BANNED: 'banned',
+		PUNISHED: 'punished',
+		NON_ELIGIBLE: 'non-eligible',
+	};
+
+	const allDelegates = await db.findAll();
+	const lastestBlock = await getBlocks({ limit: 1 });
+	const allNextForgersAddressList = nextForgers.map(forger => forger.account.address);
+	const activeNextForgersList = allNextForgersAddressList.slice(0, numActiveForgers);
+
+	const verifyIfPunished = delegate => {
+		const isPunished = delegate.pomHeights
+			.some(pomHeight => pomHeight.start <= lastestBlock.height
+				&& lastestBlock.height <= pomHeight.end);
+		return isPunished;
+	};
+
+	if (sdkVersion >= 4) allDelegates.sort(delegateComparator);
+	allDelegates.map((delegate, index) => {
+		if (sdkVersion < 4) {
+			if (activeNextForgersList.includes(delegate.account.address)) {
+				delegate.status = delegateStatus.ACTIVE;
+			} else delegate.status = delegateStatus.STANDBY;
+		} else {
+			delegate.rank = index + 1;
+			if (!delegate.isDelegate) delegate.status = delegateStatus.NON_ELIGIBLE;
+			else if (delegate.isBanned) delegate.status = delegateStatus.BANNED;
+			else if (verifyIfPunished(delegate)) delegate.status = delegateStatus.PUNISHED;
+			else if (activeNextForgersList.includes(delegate.account.address)) {
+				delegate.status = delegateStatus.ACTIVE;
+			} else delegate.status = delegateStatus.STANDBY;
+		}
+
+		return delegate;
+	});
+
+	if (allDelegates.length) await db.writeBatch(allDelegates);
+};
+
+const getNextForgers = async params => {
+	let forgers = {
+		data: [],
+		meta: {},
+		links: {},
+	};
+
+	try {
+		if (nextForgers.length) {
+			const offset = params.offset || 0;
+			const limit = params.limit || 10;
+
+			forgers.data = nextForgers.slice(offset, offset + limit);
+
+			forgers.meta.count = forgers.data.length;
+			forgers.meta.offset = offset;
+			forgers.meta.total = nextForgers.length;
+		} else throw new Error('Request Next Forgers data from Lisk Core');
+	} catch (err) {
+		logger.debug(err.message);
+
+		forgers = await coreApi.getNextForgers(params);
+		forgers.data = await BluebirdPromise.map(
+			forgers.data,
+			async forger => (await getDelegates({ address: forger.address })).data[0],
+			{ concurrency: forgers.data.length });
+		forgers.data.sort(delegateComparator);
```

Comment:
This logic can be moved to a separate function, that runs in the background.

Reply:
Done!